From 9e67ee06da67ba7091294601c24718c5bb55a6b4 Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Tue, 19 Jan 2016 13:11:06 +0000
Subject: [PATCH 08/13] usb: gadget: composite: change bcdUSB field in the
 descriptor

Due to the Device Descriptor Test failing, this change has
been implemented. It changes the bcdUSB field in the device
descriptor, so that the CHT device will be USB3.1 capable.

Note: according to the USB 3.1, both USB 3.1 Gen 1 and
USB 3.1 Gen 2 have the same bcdUSB field (Ch. 10.15.1).
bcdUSB field set to 0x310 does not indicate Type-C usage.

Change-Id: Iaacc8ffa28dce4413a0b087049455d815da72e24
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-16985
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
Reviewed-on: https://android.intel.com:443/459754
---
 drivers/usb/gadget/composite.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 11f2302..1f8c822 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -1293,7 +1293,7 @@ composite_setup(struct usb_gadget *gadget, const struct usb_ctrlrequest *ctrl)
 				cdev->gadget->ep0->maxpacket;
 			if (gadget_is_superspeed(gadget)) {
 				if (gadget->speed >= USB_SPEED_SUPER) {
-					cdev->desc.bcdUSB = cpu_to_le16(0x0300);
+					cdev->desc.bcdUSB = cpu_to_le16(0x0310);
 					cdev->desc.bMaxPacketSize0 = 9;
 				} else {
 					cdev->desc.bcdUSB = cpu_to_le16(0x0210);
-- 
1.9.1

